x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -119,16 +119,19 @@ public class Analyzer extends ParameterizedRuleExecutor<LogicalPlan, AnalyzerCon
     private static final Iterable<RuleExecutor.Batch<LogicalPlan>> rules;
 
     static {
-        var resolution = new Batch<>(
-            "Resolution",
+        var init = new Batch<>(
+            "Initialize",
+            Limiter.ONCE,
             new ResolveTable(),
             new ResolveEnrich(),
-            new ResolveFunctions(),
-            new ResolveRefs(),
-            new ImplicitCasting()
+            new ResolveLookup(),
+            new ResolveFunctions()
         );
+
+        var resolution = new Batch<>("Resolution", new ResolveRefs(), new ImplicitCasting());
```

Comment:
This is unrelated to this PR - the 'init' rules will resolve things on their first run thus it's a waste to keep using them in a loop.
The same applies for the newly introduced rule for lookup which resolves the local relationship but does NOT perform field resolution - that is left for ResolveRefs.